mkdocs/docs/bindings/c-api.md

```diff
@@ -1,9 +1,217 @@
 # C API bindings
 
-<!-- TODO(??): overview, advantages/disadvantages to using raw C APIs -->
+IREE provides a low level C API for its runtime[^1], which can be used directly or
+through higher level APIs and language bindings built on top of it.
+
+API header files are organized by runtime component:
+
+| Component header file                                                       | Overview                                                                  |
+|-----------------------------------------------------------------------------|---------------------------------------------------------------------------|
+| [iree/base/api.h](https://github.com/google/iree/blob/main/iree/base/api.h) | Core API, type definitions, ownership policies, utilities                 |
+| [iree/vm/api.h](https://github.com/google/iree/blob/main/iree/vm/api.h)     | VM APIs: loading modules, I/O, calling functions                          |
+| [iree/hal/api.h](https://github.com/google/iree/blob/main/iree/hal/api.h)   | HAL APIs: device management, synchronization, accessing hardware features |
+
+The [iree/samples/](https://github.com/google/iree/tree/main/iree/samples)
+directory demonstrates several ways to use IREE's C API.
 
 ## Prerequisites
 
-<!-- TODO(??): build from source, vcpkg -->
+To use IREE's C API, you will need to build the runtime
+[from source](../building-from-source/getting-started.md). The
+[iree-template-cpp](https://github.com/iml130/iree-template-cpp) community
+project also shows how to integrate IREE into an external project using
+CMake[^2].
+
+## Concepts
+
+### IREE VM
+
+<!-- TODO(scotttodd): VM "instances" -->
+<!-- TODO(scotttodd): VM "contexts" -->
+<!-- TODO(scotttodd): VM "modules" -->
```

Comment:
@benvanik think you could introduce the VM here for users interacting with the C API (1-2 paragraphs)?

Some snippets that could be expanded upon:

> IREE's runtime uses a Virtual Machine (VM) to ... *
> * in the general case, but this can be bypassed with [issue number]

> Pieces of functionality are imported into the VM via native modules. (talk about how even HAL drivers are loaded as modules)

> Instances and contexts are used to manage VM state

> Something about vm_list, ref types, etc. as ways to interact with the VM

---

You could also introduce the HAL concepts (drivers/devices) in the next section, but I'm also comfortable writing that.

Reply:
I'll keep working on this, starting from https://github.com/google/iree/blob/f738885923da04c7a7befe51d342473c730e8a18/iree/compiler/Dialect/VM/IR/VMBase.td#L32-L51 and https://github.com/google/iree/blob/f738885923da04c7a7befe51d342473c730e8a18/iree/compiler/Dialect/HAL/IR/HALBase.td#L32-L45